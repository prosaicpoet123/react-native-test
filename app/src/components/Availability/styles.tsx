import { StyleSheet } from 'react-native'
import { colours } from '../../config/themes'

const styles = StyleSheet.create({
  availability: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    flexDirection: 'row',
    color: colours.white,
  },
  instock: {
    backgroundColor: colours.darkGreenPrimary,
  },
  oos: {
    backgroundColor: colours.pomegranatePrimaryDark,
  },
  text: {
    paddingVertical: 2,
    paddingHorizontal: 4,
    borderRadius: 5,
    overflow: 'hidden',
    color: colours.white,
  },
})

export default styles
