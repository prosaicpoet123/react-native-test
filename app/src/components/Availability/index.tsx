import React, { FC } from 'react'
import { Text, View } from 'react-native'
import { IProduct } from '../../@Types/product'
import { parseStockStatus } from '../../utils/parseStockStatus'
import styles from './styles'

interface IAvailabilityProps {
  stock: IProduct['stock_status']
}

type IProps = IAvailabilityProps

const Availability: FC<IProps> = ({ stock }) => (
  <View style={styles.availability}>
    <Text
      style={[styles.text, stock === 'INSTOCK' ? styles.instock : styles.oos]}>
      {parseStockStatus(stock)}
    </Text>
  </View>
)

export default Availability
