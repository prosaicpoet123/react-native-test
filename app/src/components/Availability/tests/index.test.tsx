import React from 'react'
import renderer from 'react-test-renderer'
import Availability from '../'

describe('<Availability />', () => {
  it('renders correctly', () => {
    const stock_status = 'INSTOCK'
    const tree = renderer.create(<Availability stock={stock_status} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
