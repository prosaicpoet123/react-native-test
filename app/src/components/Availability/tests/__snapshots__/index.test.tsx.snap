// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Availability /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "color": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "borderRadius": 5,
          "color": "#FFFFFF",
          "overflow": "hidden",
          "paddingHorizontal": 4,
          "paddingVertical": 2,
        },
        Object {
          "backgroundColor": "#00594C",
        },
      ]
    }
  >
    In Stock
  </Text>
</View>
`;
